crates/keygen/src/keygen.rs

```diff
@@ -126,13 +130,35 @@ impl ParseSecret {
     }
 }
 
+fn wait_for_keypress() {
+    let mut single_key = [0u8];
+    std::io::stdin().read_exact(&mut single_key).unwrap();
+}
+
+fn display_string_discreetly(
+    discreet_string: &str,
+    continue_message: &str,
+) -> anyhow::Result<()> {
+    use termion::screen::IntoAlternateScreen;
+    let mut screen = std::io::stdout().into_alternate_screen()?;
+    writeln!(screen, "{discreet_string}")?;
+    screen.flush()?;
+    println!("{continue_message}");
+    wait_for_keypress();
```

Comment:
is there a way we can disable interactive cli? Otherwise this will be cumbersome to use in automation scripts.

Reply:
@Voxelot I like your suggestion. I _think_ I got it right in 5e7135e2. Basically if the stdout is a terminal the secret message, otherwise (i.e it is a pipe to another process or the stdout is redirected to a file) the normal message will be printed.

I'm attaching a video to demo how it works

https://github.com/FuelLabs/fuel-core/assets/142530682/ad8c0fe9-36ea-42cd-af19-dbe9e9519347